### PR TITLE
rename train_x_all to train_x_pde

### PR DESCRIPTION
**Pull request by g-w1**
This resolves a TODO

Was split off from #1108

### CODE REVIEW

**Comment by lululxvi** [comment]
have you double-checked if all classes using `train_x_all` has been revised?

**Comment by g-w1** [comment]
Grepping for `train_x_all` returns nothing on this branch. However, I just realized that in #1108 I use `train_x_pde`, so it would make sense to merge this pull request before that one since it depends on this one.

File: deepxde/data/pde.py [hunk 1]
```diff
@@ -100,9 +100,7 @@ def __init__(
 
         self.auxiliary_var_fn = auxiliary_var_function
 
-        # TODO: train_x_all is used for PDE losses. It is better to add train_x_pde
-        # explicitly.
```
Comment (lululxvi) [inline deepxde/data/pde.py:104]: As commented here, it is better to add `train_x_pde`, not replace `train_x_all` with `train_x_pde`. In some cases, we may actually use the meaning of `train_x_all`, but I am not sure which case. Need to look at all the code carefully.

Comment (g-w1) [reply deepxde/data/pde.py:104]: I have looked over the code, and I can't find a place where `train_x_all` does not contain PDE residual points.

If I added `train_x_pde` in addition to `train_x_all`, what would be different about them? The only thing that I can think of is that `train_x_pde` would only contain points in the domain as opposed to `train_x_all` also containing points on the boundary (or on the initial conditions in TimePDE).

What do you recommend for how to add it?

Comment (lululxvi) [reply deepxde/data/pde.py:104]: `train_x_all` contains all points (unordered, and does not have duplication). `train_x_bc` is the points for BC losses (may have have duplication).

There are cases where there are only BC losses and no PDE losses. So, it is better to add `train_x_pde`.

**Pull request closed without merging**